resources/src/ui/StateSyncModal.vue

```diff
@@ -98,18 +133,21 @@
   </cdx-dialog>
 </template>
 
-<script setup>
+<script setup lang="ts">
 import { computed } from 'vue';
-import { CdxDialog, CdxButton } from '@wikimedia/codex';
-
-const props = defineProps({
-  modelValue: { type: Boolean, default: false },
-  message: { type: String, default: '' },
-  differences: { type: Object, default: () => ({}) },
-  reconcileCommands: { type: Array, default: () => [] },
-  attemptingReconcile: { type: Boolean, default: false },
-  reconcileMessage: { type: String, default: '' },
-});
+import { CdxButton, CdxDialog } from '@wikimedia/codex';
+import { type PacksActionCommand, type StateDifference } from '../state/types.ts';
+
+interface Props {
+  modelValue: boolean;
+  message: string;
+  differences: StateDifference;
+  reconcileCommands: PacksActionCommand[];
+  attemptingReconcile: boolean;
+  reconcileMessage: string;
+}
+
+const props = defineProps<Props>();
```

Comment:
it also doesn't seem like any of these are actually optional: we only show the modal when these values exist. this syntax is the typescript way of declaring props, there's also a form with defaults: https://vuejs.org/guide/components/props#reactive-props-destructure